tools/minify.cpp

```diff
@@ -5,74 +5,50 @@
 #include <unistd.h>
 
 #include "simdjson.h"
+#include "cxxopts.hpp"
 
-// Stash the exe_name in main() for functions to use
-char* exe_name;
+cxxopts::Options options("minify", "Runs the parser against the given json files in a loop, measuring speed and other statistics.");
 
-void print_usage(std::ostream& out) {
-  out << "Usage: " << exe_name << "  [-a ARCH] <jsonfile>" << std::endl;
-  out << std::endl;
-  out << "Runs the parser against the given json files in a loop, measuring speed and other statistics." << std::endl;
-  out << std::endl;
-  out << "Options:" << std::endl;
-  out << std::endl;
-  out << "-a IMPL      - Use the given parser implementation. By default, detects the most advanced" << std::endl;
-  out << "               implementation supported on the host machine." << std::endl;
+void usage(std::string message) {
+  std::cerr << message << std::endl;
+  std::cerr << options.help() << std::endl;
+  std::cerr << "Available parser implementations:" << std::endl;
   for (auto impl : simdjson::available_implementations) {
-    out << "-a " << std::left << std::setw(9) << impl->name() << " - Use the " << impl->description() << " parser implementation." << std::endl;
+    std::cerr << "-a " << std::left << std::setw(9) << impl->name() << " - Use the " << impl->description() << " parser implementation." << std::endl;
   }
 }
 
-void exit_usage(std::string message) {
-  std::cerr << message << std::endl;
-  std::cerr << std::endl;
-  print_usage(std::cerr);
-  exit(EXIT_FAILURE);
-}
-
-
-struct option_struct {
-  char* filename{};
- 
-  option_struct(int argc, char **argv) {
-    int c;
+int main(int argc, char *argv[]) {
+  options.add_options()
+    ("a,arch", "Parser implementation (by default, detects the most advanced implementation supported on the host machine).", cxxopts::value<std::string>())
+    ("f,file", "File name.", cxxopts::value<std::string>())
+    ("h,help", "Print usage.")
+  ;
+
+  options.parse_positional({"file"});
+  auto result = options.parse(argc, argv);
+
+  if(result.count("help")) {
```

Comment:
This is sort of surprising, this is how you do a boolean parameter in cxxopts?

Reply:
Yes, this is from the example in the readme of cxxopts.
Implicitly, it's the same as checking
```c++
if(result.count("help") > 0) { ... }
```

Reply:
I agree that it is not very elegant, but that is how the authors of the library suggest to do it...https://github.com/jarro2783/cxxopts Now, there is support for boolean flags...
```
("d,debug", "Enable debugging", cxxopts::value<bool>()->default_value("false"))
...
bool debug = result["debug"].as<bool>();
```

But for "help" their examples are like so.

Reply:
Oh! I bet they do it this way because boolean options are negatable, and it doesn't make sense to have `--no-help`.